src/core/variable.jl

```diff
@@ -345,13 +345,25 @@ function variable_mc_bus_voltage_magnitude_sqr(pm::_PM.AbstractPowerModel; nw::I
     )
 
     if bounded
-        for (i,bus) in ref(pm, nw, :bus)
-            if haskey(bus, "vmin")
-                set_lower_bound.(w[i], bus["vmin"].^2)
-            end
-            if haskey(bus, "vmax")
-                set_upper_bound.(w[i], bus["vmax"].^2)
-            end
+        for i in ids(pm, nw, :bus)
+            bus = ref(pm, nw, :bus, i)
+            vmax=bus["vmax"]
+            vmin=bus["vmin"]
+            for c in 1:ncnds
+                if(vmin[c]==-Inf)
+                    a=-10000
+                else
+                    a=vmin[c]
+                end
+                if(vmax[c]==Inf)
+                    set_upper_bound.((w[i])[c], 10000.0)
+                else
+                    set_upper_bound.((w[i])[c], max(a^2, vmax[c]^2)) #this upper bound was wrong in the code
+                end
+                if(vmin[c]>0)
+                    set_lower_bound.((w[i])[c], vmin[c]^2)
+                end
+                end
```

Comment:
In v0.9 we changed things to a default of no voltage bounds (in reality, [0, Inf], in order that they still be physical), so if special voltage bounds are needed a transformation function should be used (see `/src/data_model/transformations.jl` for examples)

We should not introduce hacks into the code in this way, they will be easy to lose track of later